src/middlewared/middlewared/api/base/server/event.py

```diff
@@ -0,0 +1,19 @@
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from middlewared.main import Middleware
+
+
+class Event:
```

Comment:
Same here, `dataclass` cleans this up a bit, what do you think?

Reply:
This is not a dataclass, it will become more "active" (similar to the adjacent `Method` class) in the future